state-chain/pallets/cf-funding/src/lib.rs

```diff
@@ -351,6 +354,13 @@ pub mod pallet {
 
 			let mut restricted_balances = RestrictedBalances::<T>::get(&account_id);
 
+			// ensure that all restricted balances are above MinimumFunding, to avoid
+			// situations where it is not possible to redeem max
+			ensure!(
+				restricted_balances.values().all(|v| *v >= MinimumFunding::<T>::get()),
+				Error::<T>::RestrictedBalanceBelowMinimumFunding
+			);
```

Comment:
I think we need to do the check *after* we have deducted the redemption amount. 

The problem was that there was dust remaining in the account after the withdrawal.

Reply:
agree, plz see updated code